File: src/main/java/cloud/cosmin/checklister/dao/ItemEntity.java
package cloud.cosmin.checklister.dao;

import org.hibernate.annotations.DynamicInsert;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.util.Objects;
import java.util.UUID;

@Entity
@DynamicInsert
@Table(name = "item")
public class ItemEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private int rank;

    @Column(nullable = false)
    private String contentType;

    @ManyToOne
    @JoinColumn(nullable = false)
    private ListEntity list;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getContentType() {
        return contentType;
    }

    public void setContentType(String contentType) {
        this.contentType = contentType;
    }

    public int getRank() {
        return rank;
    }

    public void setRank(int rank) {
        this.rank = rank;
    }

    public ListEntity getList() {
        return list;
    }

    public void setList(ListEntity list) {
        this.list = list;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ItemEntity that = (ItemEntity) o;
        return rank == that.rank &&
                Objects.equals(id, that.id) &&
                Objects.equals(content, that.content) &&
                Objects.equals(contentType, that.contentType) &&
                Objects.equals(list, that.list);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, content, rank, contentType, list);
    }
}

File: src/main/java/cloud/cosmin/checklister/dao/ItemEntity.kt
package cloud.cosmin.checklister.dao

import org.hibernate.annotations.DynamicInsert

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table
import java.util.Objects
import java.util.UUID

@Entity
@DynamicInsert
@Table(name = "item")
class ItemEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: UUID? = null

    @Column(nullable = false)
    var content: String? = null

    @Column(nullable = false)
    var rank: Int = 0

    @Column(nullable = false)
    var contentType: String? = null

    @ManyToOne
    @JoinColumn(nullable = false)
    var list: ListEntity? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ItemEntity?
        return rank == that!!.rank &&
                id == that.id &&
                content == that.content &&
                contentType == that.contentType &&
                list == that.list
    }

    override fun hashCode(): Int {
        return Objects.hash(id, content, rank, contentType, list)
    }
}
